testmod-common/src/main/java/me/shedaniel/architectury/test/TestMod.java

```diff
@@ -42,7 +45,9 @@ public static void initialize() {
         TestGameRules.init();
         TestTags.initialize();
         TestTrades.init();
-        if (Platform.getEnvironment() == Env.CLIENT)
+        if (Platform.getEnvironment() == Env.CLIENT) {
             TestKeybinds.initialize();
+            EntityRenderers.register(TestEntity.TYPE, MinecartRenderer<TestEntity>::new);
```

Comment:
This will crash if the test mod is run on a dedicated server.

Reply:
I just fired one up and it doesn't, probably because of the side check two lines above

Reply:
Huh, I would expect the class loader to crash because the method references `MinecartRenderer` (which is client-only), but the method reference might prevent that.

Reply:
Yeah the method is only references when it ever reaches that Class call, but it never does on a dedicated server, because of the client check.
I use this principle a lot, like I never use forge proxies, but just checking if the mod is loaded client or serverside and then go into a different method for handling the side. I needed to understand the bytecode first, but now it is very obvious for me.